### PR TITLE
Develop fix cumulative reward

### PR DESCRIPTION
**Pull request by vincentpierre**
* Split the way cummulative rewards and episode length are counted
 * The reward is appended at each step to the cumulative reward
 * The episode count is ONLY incremented when d_t+1 is false

### CODE REVIEW

File: python/unitytrainers/ppo/trainer.py [hunk 1]
```diff
@@ -254,10 +254,10 @@ def add_experiences(self, curr_all_info: AllBrainInfo, next_all_info: AllBrainIn
                     self.training_buffer[agent_id]['action_probs'].append(a_dist[idx])
                     self.training_buffer[agent_id]['value_estimates'].append(value[idx][0])
 
-                if not next_info.local_done[next_idx]:
                     if agent_id not in self.cumulative_rewards:
```
Comment (awjuliani) [inline python/unitytrainers/ppo/trainer.py]: Maybe move these three lines up to ensure good visual interpretability.
